mountpoint-s3/src/main.rs

```diff
@@ -597,34 +561,25 @@ fn mount(args: CliArgs) -> anyhow::Result<FuseSession> {
 
     #[cfg(feature = "caching")]
     {
-        use mountpoint_s3::data_cache::{CacheLimit, DiskDataCache};
+        use mountpoint_s3::data_cache::{CacheLimit, DiskDataCache, DiskDataCacheConfig};
         use mountpoint_s3::fs::CacheConfig;
         use mountpoint_s3::prefetch::caching_prefetch;
 
-        if args.enable_metadata_caching {
-            // TODO: Review default for TTL
-            let metadata_cache_ttl = args.metadata_cache_ttl.unwrap_or(Duration::from_secs(3600));
+        if let Some(path) = args.cache {
+            let metadata_cache_ttl = args.metadata_ttl.unwrap_or(Duration::from_secs(1));
             filesystem_config.cache_config = CacheConfig {
                 serve_lookup_from_cache: true,
                 dir_ttl: metadata_cache_ttl,
                 file_ttl: metadata_cache_ttl,
             };
-        }
-
-        if let Some(path) = args.data_caching_directory {
-            let limit = {
-                if let Some(max_size_in_mb) = args.data_cache_size_limit {
-                    let max_size = (max_size_in_mb * 1024 * 1024) as usize;
-                    CacheLimit::TotalSize { max_size }
-                } else if let Some(per_cent) = args.data_cache_free_space {
-                    let min_ratio = (per_cent as f64) * 0.01;
-                    CacheLimit::AvailableSpace { min_ratio }
-                } else {
-                    CacheLimit::Unbounded
-                }
-            };
 
-            let cache = DiskDataCache::new(path, args.data_cache_block_size, limit);
+            let mut cache_config = DiskDataCacheConfig::default();
+            if let Some(max_size_in_mb) = args.max_cache_size {
+                cache_config.limit = CacheLimit::TotalSize {
+                    max_size: (max_size_in_mb * 1_000_000) as usize,
+                };
+            }
```

Comment:
Why megabytes over mebibytes?

Reply:
To match the flag description. Happy to change, but then would we also have something like this?
```
--max-cache-size <MiB>     Maximum size of the cache directory in MiB [default: preserve 1% of available space]
```

Reply:
Yeah, we'd need to change both

Reply:
Let's use MiB.